src/pool/cli.c

```diff
@@ -274,21 +274,121 @@ dc_pool_map_update(struct dc_pool *pool, struct pool_map *map, bool connect)
 	return rc;
 }
 
+static void
+pool_print_range_list(struct dc_pool *pool, d_rank_range_list_t *list, bool enabled)
+{
+	const size_t	MAXBYTES = 512;
+	char		line[MAXBYTES];
+	char	       *linepos = line;
+	int		ret;
+	unsigned int	written = 0;
+	unsigned int	remaining = MAXBYTES;
+	int		i;
+
+	ret = snprintf(linepos, remaining, DF_UUID": %s ranks: ", DP_UUID(pool->dp_pool),
+		       enabled ? "ENABLED" : "DISABLED");
+	if (ret < 0)
+		goto err;
+	written += ret;
+	remaining -= ret;
+	linepos += ret;
+
+	for (i = 0; i < list->rrl_nr; i++) {
+		uint32_t	lo = list->rrl_ranges[i].lo;
+		uint32_t	hi = list->rrl_ranges[i].hi;
+		bool		lastrange = (i == (list->rrl_nr - 1));
+
+		if (lo == hi)
+			ret = snprintf(linepos, remaining, "%u%s", lo, lastrange ? "" : ",");
+		else
+			ret = snprintf(linepos, remaining, "%u-%u%s", lo, hi, lastrange ? "" : ",");
+		if (ret < 0)
+			goto err;
+		written += ret;
+		remaining -= ret;
+		linepos += ret;
```

Comment:
The return value of snprintf may be >= remaining if the buffer isn't large enough.